diff -Naur pdfmixtool-v0.6.orig/src/pdf_edit_lib/pdf_writer.cpp pdfmixtool-v0.6/src/pdf_edit_lib/pdf_writer.cpp
--- pdfmixtool-v0.6.orig/src/pdf_edit_lib/pdf_writer.cpp	2020-07-04 15:23:18.179265633 -0400
+++ pdfmixtool-v0.6/src/pdf_edit_lib/pdf_writer.cpp	2020-07-04 15:23:37.346343765 -0400
@@ -465,7 +465,7 @@
                             QPDFObjectHandle::Rectangle(
                                 0, 0,
                                 page_width, page_height),
-                            false, true, true);
+                            false);
 
                 if (! content.empty())
                 {
