--- a/Makefile	2015-07-29 17:57:24.000000000 -0300
+++ b/Makefile	2016-07-13 21:42:35.297421714 -0400
@@ -64,8 +64,8 @@
 
 install-openrc:
 	$(Q)echo -e '\033[1;32mInstalling openrc files...\033[0m'
-	$(INSTALL_DIR) "$(DESTDIR)$(DOCDIR)"
-	$(INSTALL_SCRIPT) init/psd.openrc "$(DESTDIR)$(DOCDIR)/psd.openrc"
+	$(INSTALL_DIR) "$(DESTDIR)$(INITDIR_UPSTART)"
+	$(INSTALL_SCRIPT) init/psd.openrc "$(DESTDIR)$(INITDIR_UPSTART)/psd"
 	[ -f "$(DESTDIR)$(CONFDIR)/psd.conf" ] || $(INSTALL_DATA) common/psd.conf "$(DESTDIR)$(CONFDIR)/psd.conf"
 
 install-systemd:
@@ -115,7 +115,8 @@
 	$(RM) "$(DESTDIR)$(CRONDIR)/psd-update"
 
 uninstall-openrc:
-	$(RM) "$(DESTDIR)$(DOCDIR)/psd.openrc"
+	$(RM) "$(DESTDIR)$(CONFDIR)/psd.conf"
+	$(RM) "$(DESTDIR)$(INITDIR_UPSTART)/psd"
 
 uninstall-systemd:
 	$(RM) "$(DESTDIR)$(CONFDIR)/psd.conf"
