--- a/common/profile-sync-daemon.in	2017-05-15 00:51:33.765195481 +0200
+++ b/common/profile-sync-daemon.in	2017-05-14 12:59:14.875445714 +0200
@@ -548,7 +548,7 @@
 				# check that the LAST DIRECTORY in the full path is unique
 				unique_count=$(echo ${DIRArr[@]##*/} |
 				sed 's/ /\n/g' | sort -u | wc -l)
-				if [[ ${#DIRArr[@]##*/} -eq $unique_count ]]; then
+				if [[ ${#DIRArr[@]} -eq $unique_count ]]; then
 					# no problems so do nothing
 					return
 				else
@@ -570,7 +570,7 @@
 				# check that the LAST DIRECTORY in the full path is unique
 				unique_count=$(echo ${DIRArr[@]##*/} |
 				sed 's/ /\n/g' | sort -u | wc -l)
-				if [[ ${#DIRArr[@]##*/} -eq $unique_count ]]; then
+				if [[ ${#DIRArr[@]} -eq $unique_count ]]; then
 					# no problems so do nothing
 					return
 				else
