--- lightdm-another-gtk-greeter-settings-1.0.0/data/Makefile.am	2015-06-06 01:33:10.851553722 -0700
+++ lightdm-another-gtk-greeter-settings-1.0.0/data/Makefile.am	2010-06-06 01:34:52.261555836 -0700
@@ -10,8 +10,13 @@
 
 UPDATE_DESKTOP = update-desktop-database $(datadir)/applications || :
 
-install-data-hook: 
-	$(UPDATE_DESKTOP)
-uninstall-hook: 
-	$(UPDATE_DESKTOP)
-
+install-data-hook: update-desktop-database
+uninstall-hook: update-desktop-database
+update-desktop-database:
+	@-if test -z "$(DESTDIR)"; then \
+		echo "Updating Desktop Database."; \
+		$(UPDATE_DESKTOP); \
+	else \
+		echo "*** Desktop Database not updated.  After (un)install, run this:"; \
+		echo "***   update-desktop-database $(desktopdir)"; \
+	fi

