diff -Naur a/fontconfig/56-twemoji-color.conf b/fontconfig/56-twemoji-color.conf
--- a/fontconfig/56-twemoji-color.conf	2017-09-01 00:23:23.091538971 -0300
+++ b/fontconfig/56-twemoji-color.conf	2017-09-01 00:33:28.751550884 -0300
@@ -26,98 +26,29 @@
 fc-match -s monospace
 -->
 <fontconfig>
-  <match target="font">
-    <!-- If the requested font is Bitstream Vera Serif -->
-    <test name="family" compare="eq">
-        <string>Bitstream Vera Serif</string>
-    </test>
-    <!-- Replace the entire match list with Bitstream Vera Serif alone -->
-    <edit name="family" mode="assign_replace">
-      <string>Bitstream Vera Serif</string>
-    </edit>
-    <!-- Assign the serif family -->
-    <edit name="family" mode="append_last">
-      <string>serif</string>
-    </edit>
-  </match>
+	<alias>
+		<family>sans-serif</family>
+		<prefer>
+			<family>Twitter Color Emoji</family>
+			<family>Bitstream Vera Sans</family>
+		</prefer>
+	</alias>
+	<alias>
+		<family>serif</family>
+		<prefer>
+			<family>Twitter Color Emoji</family>
+			<family>Bitstream Vera Serif</family>
+		</prefer>
+	</alias>
+
+	<alias>
+		<family>monospace</family>
+		<prefer>
+			<family>Twitter Color Emoji</family>
+			<family>Bitstream Vera Sans Mono</family>
+		</prefer>
+	</alias>
 
-  <match>
-    <!-- If the requested font is serif -->
-    <test qual="any" name="family">
-        <string>serif</string>
-    </test>
-    <!-- Make Bitstream Vera Serif the first result -->
-    <edit name="family" mode="prepend_first">
-      <string>Bitstream Vera Serif</string>
-    </edit>
-    <!-- Followed by Twitter Color Emoji -->
-    <edit name="family" mode="prepend_first">
-      <string>Twitter Color Emoji</string>
-    </edit>
-  </match>
-
-  <match target="font">
-    <!-- If the requested font is Bitstream Vera Sans -->
-    <test name="family" compare="eq">
-        <string>Bitstream Vera Sans</string>
-    </test>
-    <!-- Replace the entire match list with Bitstream Vera Sans alone -->
-    <edit name="family" mode="assign_replace">
-      <string>Bitstream Vera Sans</string>
-    </edit>
-    <!-- Assign the sans-serif family -->
-    <edit name="family" mode="append_last">
-      <string>sans-serif</string>
-    </edit>
-  </match>
-
-  <match target="pattern">
-    <!-- If the requested font is sans-serif -->
-    <test qual="any" name="family">
-        <string>sans-serif</string>
-    </test>
-    <!-- Make Bitstream Vera Sans the first result -->
-    <edit name="family" mode="prepend_first">
-      <string>Bitstream Vera Sans</string>
-    </edit>
-    <!-- Followed by Twitter Color Emoji -->
-    <edit name="family" mode="prepend_first">
-      <string>Twitter Color Emoji</string>
-    </edit>
-  </match>
-
-  <match target="font">
-    <!-- If the requested font is Bitstream Vera Sans Mono -->
-    <test name="family" compare="eq">
-        <string>Bitstream Vera Sans Mono</string>
-    </test>
-    <!-- Replace the entire match list with Bitstream Vera Sans Mono alone -->
-    <edit name="family" mode="assign_replace">
-      <string>Bitstream Vera Sans Mono</string>
-    </edit>
-    <!-- Assign the monospace family last -->
-    <edit name="family" mode="append_last">
-      <string>monospace</string>
-    </edit>
-  </match>
-
-  <match target="pattern">
-    <!-- If the requested font is monospace -->
-    <test qual="any" name="family">
-        <string>monospace</string>
-    </test>
-    <!--
-    Make Bitstream Vera Sans Mono the first result
-    Note: If you want a different monospace font, this is where you change it.
-    -->
-    <edit name="family" mode="prepend_first">
-      <string>Bitstream Vera Sans Mono</string>
-    </edit>
-    <!-- Followed by Twitter Color Emoji -->
-    <edit name="family" mode="prepend_first">
-      <string>Twitter Color Emoji</string>
-    </edit>
-  </match>
 
   <!-- Add emoji generic family -->
   <alias binding="strong">
