From 9741c39382a3f6e4c03eac6905a49794d07c465a Mon Sep 17 00:00:00 2001
From: Ivan Malison <IvanMalison@gmail.com>
Date: Thu, 14 Jun 2018 23:48:32 -0700
Subject: [PATCH] Fix issues with newer versions of gtk

---
 src/main.c | 51 +++++++++++++++++++++++----------------------------
 1 file changed, 23 insertions(+), 28 deletions(-)

diff --git a/src/main.c b/src/main.c
index 1743233..0fd2aec 100644
--- a/src/main.c
+++ b/src/main.c
@@ -24,6 +24,8 @@
 #include <config.h>
 #endif
 
+#define _GNU_SOURCE
+
 #include <glib.h>
 #include <stdlib.h>
 #include <gtk/gtk.h>
@@ -541,7 +543,7 @@ gboolean selected_by_input(const GtkWidget *history_menu, const GdkEventKey *eve
     return FALSE;
   }
 
-  if (event->keyval == GDK_Down || event->keyval == GDK_Up) {
+  if (event->keyval == GDK_KEY_Down || event->keyval == GDK_KEY_Up) {
     return FALSE;
   }
 
@@ -549,7 +551,7 @@ gboolean selected_by_input(const GtkWidget *history_menu, const GdkEventKey *eve
     append_to_input_buffer(event->string);
 
   GtkMenuShell* menu = (GtkMenuShell *) history_menu;
-  GList* element = menu->children;
+  GList* element = gtk_container_get_children(menu);
   GtkMenuItem *menu_item, *first_match = 0;
 
   const gchar* menu_label;
@@ -578,69 +580,62 @@ gboolean selected_by_input(const GtkWidget *history_menu, const GdkEventKey *eve
 
   if (first_match && match_count != prefs.items_menu) {
     gtk_menu_item_select(first_match);
-    menu->active_menu_item = (GtkWidget *) first_match;
+	gtk_menu_shell_select_item(menu, first_match);
     return TRUE;
   }
   return FALSE;
 }
 
 gboolean selected_by_digit(const GtkWidget *history_menu, const GdkEventKey *event) {
+  int selection = -1;
   switch (event->keyval) {
     case XK_1:
     case XK_KP_1:
-      item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(0));
-      gtk_widget_destroy(history_menu);
+	  selection = 1;
       break;
     case XK_2:
     case XK_KP_2:
-      item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(1));
-      gtk_widget_destroy(history_menu);
+	  selection = 2;
       break;
     case XK_3:
     case XK_KP_3:
-      item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(2));
-      gtk_widget_destroy(history_menu);
+	  selection = 3;
       break;
     case XK_4:
     case XK_KP_4:
-      item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(3));
-      gtk_widget_destroy(history_menu);
+	  selection = 4;
       break;
     case XK_5:
     case XK_KP_5:
-      item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(4));
-      gtk_widget_destroy(history_menu);
+	  selection = 5;
       break;
     case XK_6:
     case XK_KP_6:
-      item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(5));
-      gtk_widget_destroy(history_menu);
+	  selection = 6;
       break;
     case XK_7:
     case XK_KP_7:
-      item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(6));
-      gtk_widget_destroy(history_menu);
+	  selection = 7;
       break;
     case XK_8:
     case XK_KP_8:
-      item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(7));
-      gtk_widget_destroy(history_menu);
+	  selection = 8;
       break;
     case XK_9:
     case XK_KP_9:
-      item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(8));
-      gtk_widget_destroy(history_menu);
+	  selection = 9;
       break;
     case XK_0:
     case XK_KP_0:
-      item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(9));
-      gtk_widget_destroy(history_menu);
+	  selection = 0;
       break;
-
-    default:
-      return FALSE;
   }
-  return TRUE;
+  if (selection >= 0) {
+	item_selected((GtkMenuItem*)history_menu, GINT_TO_POINTER(selection));
+	gtk_widget_destroy(history_menu);
+	return TRUE;
+  }
+  return FALSE;
 }
 
 static void toggle_offline_mode() {
@@ -727,7 +722,7 @@ static GtkWidget *create_history_menu(GtkWidget *history_menu) {
 				element_number--;
 			else
 				element_number++;
-				element_number_small++;
+			element_number_small++;
 		}
 		/* Cleanup */
 		g_free(primary_temp);
