diff -Naur a/asemantools/qtsingleapplication/qtlocalpeer.cpp b/asemantools/qtsingleapplication/qtlocalpeer.cpp
--- a/asemantools/qtsingleapplication/qtlocalpeer.cpp	2017-08-08 00:07:41.969436751 -0400
+++ b/asemantools/qtsingleapplication/qtlocalpeer.cpp	2017-08-08 00:08:16.699436804 -0400
@@ -48,6 +48,7 @@
 #include "qtlocalpeer.h"
 #include <QtCore/QCoreApplication>
 #include <QtCore/QTime>
+#include <QtCore/QDataStream>
 
 #if defined(Q_OS_WIN)
 #include <QtCore/QLibrary>
