diff --git a/build/tasks/build-task.coffee b/build/tasks/build-task.coffee
index c9964b4..74d8881 100644
--- a/build/tasks/build-task.coffee
+++ b/build/tasks/build-task.coffee
@@ -15,8 +15,6 @@ module.exports = (grunt) ->
 
     if process.platform is 'darwin'
       cp 'atom-shell/Atom.app', shellAppDir, filter: /default_app/
-    else
-      cp 'atom-shell', shellAppDir, filter: /default_app/
 
     mkdir appDir
 
