diff -Naur a/meson.build b/meson.build
--- a/meson.build	2017-07-07 20:01:20.743590541 -0400
+++ b/meson.build	2017-07-07 20:00:41.253591198 -0400
@@ -7,7 +7,12 @@
 project_id = 'org.gnome.Lollypop'
 
 message('Looking for dependencies')
-dependency('python3')
+python_bin = python.find_python()
+if python_bin.found()
+  message('Found python3 binary')
+else
+  error('No valid python3 binary found')
+endif
 dependency('glib-2.0')
 dependency('gobject-introspection-1.0', version: '>=1.35.9')
 dependency('gtk+-3.0', version :'>=3.14')
@@ -36,6 +41,7 @@
 )
 
 message('Preparing init file')
+run_command(['chmod', '+x', 'lollypop.in'])
 configure_file(
   input: 'lollypop.in',
   output: 'lollypop',
@@ -43,6 +49,7 @@
   install_dir: 'bin'
 )
 
+run_command(['chmod', '+x', 'lollypop-cli.in'])
 configure_file(
   input: 'lollypop-cli.in',
   output: 'lollypop-cli',
@@ -50,6 +57,7 @@
   install_dir: 'bin'
 )
 
+run_command(['chmod', '+x', 'rhythmbox2lollypop.in'])
 configure_file(
   input: 'rhythmbox2lollypop.in',
   output: 'rhythmbox2lollypop',
