--- a/wscript	2016-04-23 15:58:44.000000000 -0300
+++ b/wscript	2016-10-09 19:27:41.572144143 -0300
@@ -99,6 +99,7 @@
     conf.env['PLUGINS_DIR'] = os.path.join(conf.env['LIBDIR'], APPNAME, 'plugins')
     conf.define('PLUGINS_DATA_DIR', os.path.join(conf.env['DATADIR'], APPNAME, 'plugins'))
     conf.define('TEST_DATA_DIR', conf.path.abspath() + '/tests/data/')
+    conf.env['LINKFLAGS'] += ['-fuse-ld=bfd']
       
     # set 'default' variant
     conf.define ('DEBUG', 0)
