diff -Naur a/wscript b/wscript
--- a/wscript	2017-07-24 22:31:28.777184993 -0400
+++ b/wscript	2017-07-24 22:32:26.377186126 -0400
@@ -119,6 +119,7 @@
     # in any case we need to ignore warnings as C-code is generated
     conf.env['CFLAGS'] += ['-w']
     conf.env['VALAFLAGS'] = ['--disable-assert']
+    conf.env['LINKFLAGS'] += ['-fuse-ld=bfd']
 
     # set some debug relevant config values
     if Options.options.debug:
