--- a/waflib/Tools/glib2.py	2016-10-09 21:54:08.932146663 -0300
+++ b/waflib/Tools/glib2.py	2016-10-09 21:54:30.962146670 -0300
@@ -329,7 +329,6 @@
 			self.bld.install_files (self.env['GSETTINGSSCHEMADIR'], install_files)
 
 			if not hasattr(self.bld, '_compile_schemas_registered'):
-				self.bld.add_post_fun (compile_schemas_callback)
 				self.bld._compile_schemas_registered = True
 
 class glib_validate_schema(Task.Task):
