From 86429b91aea29535be6598d53b022f8e7611fc90 Mon Sep 17 00:00:00 2001
From: Dustin Falgout <dustin@falgout.us>
Date: Mon, 3 Apr 2017 23:13:58 -0500
Subject: [PATCH] Patch to support webkit2gtk 2.16+. Fixes #128

---
 meson.build             | 13 +++++++++----
 src/webkit2-extension.c |  6 ++++++
 2 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 258368e..dab7b3c 100644
--- a/meson.build
+++ b/meson.build
@@ -1,4 +1,4 @@
-project('lightdm-webkit2-greeter', 'c', version: '2.2.3', license: 'GPL-3')
+project('lightdm-webkit2-greeter', 'c', version: '2.2.4', license: 'GPL-3')
 
 
 # ================================== #
@@ -28,10 +28,11 @@ webkit2_webext  = dependency('webkit2gtk-web-extension-4.0', version: '>=2.12')
 greeter_deps = [dbus_glib, gtk3, webkit2, x11]
 webext_deps = [webkit2_webext, lightdm_gobject]
 
-has_webkitgtk_2_14 = webkit2.version().version_compare('>=2.14')
+has_webkitgtk_2_14   = webkit2.version().version_compare('>=2.14')
 has_webkitgtk_2_14_4 = webkit2.version().version_compare('>=2.14.4')
-has_lightdm_1_19_2 = lightdm_gobject.version().version_compare('>=1.19.2')
-has_gtk_3_22 = gtk3.version().version_compare('>=3.22')
+has_webkitgtk_2_16   = webkit2.version().version_compare('>=2.16')
+has_lightdm_1_19_2   = lightdm_gobject.version().version_compare('>=1.19.2')
+has_gtk_3_22         = gtk3.version().version_compare('>=3.22')
 
 # =================================== #
 # ------->>> Configuration <<<------- #
@@ -67,6 +68,10 @@ if has_webkitgtk_2_14_4
   conf.set('HAS_WEBKITGTK_2_14_4', 'TRUE')
 endif
 
+if has_webkitgtk_2_16
+  conf.set('HAS_WEBKITGTK_2_16', 'TRUE')
+endif
+
 if has_lightdm_1_19_2
   conf.set('HAS_LIGHTDM_1_19_2', has_lightdm_1_19_2)
 endif
diff --git a/src/webkit2-extension.c b/src/webkit2-extension.c
index 4eaf757..e18ab83 100644
--- a/src/webkit2-extension.c
+++ b/src/webkit2-extension.c
@@ -43,7 +43,13 @@
 #include <gtk/gtk.h>
 #include <glib/gi18n.h>
 #include <glib/gstdio.h>
+
+#ifdef HAS_WEBKITGTK_2_16
+#include <webkitdom/WebKitDOMDOMWindow.h>
+#else
 #include <webkitdom/WebKitDOMCustom.h>
+#endif
+
 #include <webkit2/webkit-web-extension.h>
 #include <JavaScriptCore/JavaScript.h>
 #include <lightdm.h>
