diff -Naur work/internal/gumbo/CMakeLists.txt workmodified/internal/gumbo/CMakeLists.txt
--- work/internal/gumbo/CMakeLists.txt	2017-01-04 13:45:27.000000000 -0300
+++ workmodified/internal/gumbo/CMakeLists.txt	2017-03-31 11:35:33.840000843 -0300
@@ -40,4 +40,5 @@
 
 if( UNIX AND NOT APPLE )
 	set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99" )
+    install( TARGETS sigilgumbo DESTINATION ${CMAKE_INSTALL_PREFIX}/${LIBDIR}/sigil/ )
 endif()
diff -Naur work/src/CMakeLists.txt workmodified/src/CMakeLists.txt
--- work/src/CMakeLists.txt	2017-02-02 12:35:19.000000000 -0300
+++ workmodified/src/CMakeLists.txt	2017-03-31 11:37:17.770000873 -0300
@@ -962,12 +962,10 @@
     endif()
 
     # Standard Linux 'make install'
-    install( TARGETS sigil DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/sigil/ )
+    install( TARGETS sigil DESTINATION ${CMAKE_INSTALL_PREFIX}/${LIBDIR}/sigil/ )
     install( PROGRAMS ${LINUX_LAUNCH_INSTALL_SCRIPT_CONFIGURED} DESTINATION ${CMAKE_INSTALL_PREFIX}/bin/ RENAME ${PROJECT_NAME} )
     if( NOT USE_SYSTEM_LIBS OR NOT HUNSPELL_FOUND )
-        install( PROGRAMS ${HUNSPELL_NAME} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/sigil/ )
     endif()
-    install( PROGRAMS ${GUMBO_NAME} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/sigil/ )
     if( BUILD_PATCHED_LIBXML2 )
         install( PROGRAMS ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libsigilxml2.so.2.9.4 DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/sigil/ )
         install( FILES ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/libxml2.so.2 DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/sigil/ )
