From 2bf01ff590359932f28f79272781eedd541af8e9 Mon Sep 17 00:00:00 2001
From: Renee D'Netto <rdnetto@gmail.com>
Date: Fri, 12 Jun 2020 21:56:17 +1000
Subject: [PATCH] Don't download file

---
 third-party/CMakeLists.txt | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/third-party/CMakeLists.txt b/third-party/CMakeLists.txt
index 0026f59..4489070 100644
--- a/third-party/CMakeLists.txt
+++ b/third-party/CMakeLists.txt
@@ -15,9 +15,6 @@ set(MSGPACK_SHA256 ff865a36bad5c72b8e7ebc4b7cf5f27a820fce4faff9c571c1791e3728355
 
 message(STATUS "Downloading Msgpack...")
 set(MSGPACK_TARBALL msgpack-${MSGPACK_VERSION}.tar.gz)
-file(DOWNLOAD ${MSGPACK_URL} ${CMAKE_CURRENT_SOURCE_DIR}/${MSGPACK_TARBALL}
-	INACTIVITY_TIMEOUT 30
-	EXPECTED_HASH SHA256=${MSGPACK_SHA256})
 execute_process(COMMAND ${CMAKE_COMMAND} -E tar xfz ${MSGPACK_TARBALL}
 	WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
 	RESULT_VARIABLE rv)
-- 
2.25.3

