diff -Naur a/Makefile b/Makefile
--- a/Makefile	2018-11-17 01:01:04.129999830 -0300
+++ b/Makefile	2018-11-17 01:02:06.306584411 -0300
@@ -60,9 +60,6 @@
 		install -d -m 755 $(MANDIR)/$$i; \
 		install -m 644 $$i/* $(MANDIR)/$$i; \
 	done
-	install -m 644 LEEME.extra $(MANDIR)
-	install -m 644 PROYECTO $(MANDIR)
-	install -m 644 PAQUETES $(MANDIR)
 
 # Tal vez alguien también querría mirar en el directorio /var/catman/cat2 o
 # algo por el estilo. El problema es que la posición de las páginas cat
