diff -Naur a/common/profile-cleaner.in b/common/profile-cleaner.in
--- a/common/profile-cleaner.in	2020-05-12 21:11:28.104357941 -0400
+++ b/common/profile-cleaner.in	2020-05-12 21:52:44.818506786 -0400
@@ -318,6 +318,14 @@
     do_dbbased
     exit 0
     ;;
+   V|v)
+    for name in vivaldi vivaldi-snapshot; do
+        export name
+        prepath="$XDG_CONFIG_HOME"/$name
+        [[ -d "$XDG_CONFIG_HOME"/$name ]] && do_chromebased
+    done
+    exit 0
+    ;;
   *)
     echo -e " ${BLD}$0 ${NRM}${GRN}{browser abbreviation}${NRM}"
     echo
@@ -339,6 +347,7 @@
     echo -e "   ${BLD}s) ${GRN}s${NRM}${BLD}eamonkey${NRM}"
     echo -e "   ${BLD}t) ${GRN}t${NRM}${BLD}hunderbird${NRM}"
     echo -e "  ${BLD}to) ${GRN}to${NRM}${BLD}rbrowser${NRM}"
+    echo -e "   ${BLD}v) ${GRN}v${NRM}${BLD}ivaldi (stable and snapshot)${NRM}"
     echo
     echo -e "   ${BLD}p) ${GRN}p${NRM}${BLD}aths${NRM}"
     exit 0
diff -Naur a/common/zsh-completion b/common/zsh-completion
--- a/common/zsh-completion	2020-05-12 21:11:28.104357941 -0400
+++ b/common/zsh-completion	2020-05-12 21:56:40.483151481 -0400
@@ -8,6 +8,7 @@
 	'f:Firefox (stable and beta)'
 	'gc:Google-Chrome (stable, beta, and dev)'
 	'o:Opera (stable, next, and developer)'
+    'v:Vivaldi (stable and snapshot)'
 	'ck:ConKeror'
 	'h:Heftigs aurora'
 	'i:Icecat'
diff -Naur a/doc/pc.1 b/doc/pc.1
--- a/doc/pc.1	2020-05-12 21:11:28.104357941 -0400
+++ b/doc/pc.1	2020-05-12 21:55:40.435806500 -0400
@@ -73,6 +73,10 @@
 Clean tor-browser's profile
 .TP
 .B
+v
+Clean vivaldi's profile
+.TP
+.B
 s
 Clean seamonkey's \fBprofile\fP(s)*
 .TP
diff -Naur a/README.md b/README.md
--- a/README.md	2020-05-12 21:11:28.104357941 -0400
+++ b/README.md	2020-05-12 21:54:39.820490407 -0400
@@ -18,6 +18,7 @@
 * Seamonkey
 * Thunderbird
 * Tor-browser
+* Vivaldi (stable and snapshot)
 
 ## Packages by Distro
 * ![logo](http://www.monitorix.org/imgs/archlinux.png "arch logo")Arch via AUR Package: https://aur.archlinux.org/packages/profile-cleaner
