diff -Naur a/common/profile-cleaner.in b/common/profile-cleaner.in
--- a/common/profile-cleaner.in	2020-05-16 12:14:59.033740280 -0400
+++ b/common/profile-cleaner.in	2020-05-16 12:19:33.217328995 -0400
@@ -172,10 +172,10 @@
 
 case "$1" in
   B|b)
-    for name in BraveSoftware Brave-Browser; do
+    for name in Brave-Browser Brave-Browser-Dev Brave-Browser-Beta Brave-Browser-Nightly; do
       export name
-      prepath="$XDG_CONFIG_HOME"/$name
-      if [[ -d "$XDG_CONFIG_HOME"/$name ]]; then
+      prepath="$XDG_CONFIG_HOME"/BraveSoftware/$name
+      if [[ -d "$XDG_CONFIG_HOME"/BraveSoftware/$name ]]; then
         do_chromebased
       else
         do_error_for_chromebased
@@ -331,7 +331,7 @@
     *)
       echo -e " ${BLD}$0 ${NRM}${GRN}{browser abbreviation}${NRM}"
       echo
-      echo -e "   ${BLD}b) ${GRN}b${NRM}${BLD}rave${NRM}"
+      echo -e "   ${BLD}b) ${GRN}b${NRM}${BLD}rave (stable, beta, dev and nightly)${NRM}"
       echo -e "   ${BLD}c) ${GRN}c${NRM}${BLD}hromium (stable, beta, and dev)${NRM}"
       echo -e "   ${BLD}f) ${GRN}f${NRM}${BLD}irefox (stable and beta)${NRM}"
       echo -e "  ${BLD}fa) ${GRN}fa${NRM}${BLD}lkon${NRM}"
diff -Naur a/common/zsh-completion b/common/zsh-completion
--- a/common/zsh-completion	2020-05-16 12:14:59.033740280 -0400
+++ b/common/zsh-completion	2020-05-16 12:19:52.534271699 -0400
@@ -4,7 +4,7 @@
 	local -a options
 
 	options=(
-		'b:Brave'
+		'b:Brave (stable, beta, dev and nightly)'
 		'c:Chromium (stable, beta, and dev)'
 		'ck:ConKeror'
 		'f:Firefox (stable and beta)'
diff -Naur a/doc/pc.1 b/doc/pc.1
--- a/doc/pc.1	2020-05-16 12:14:59.033740280 -0400
+++ b/doc/pc.1	2020-05-16 12:20:22.534300267 -0400
@@ -130,7 +130,7 @@
 .SH SUPPORTED BROWSERS AND EMAIL CLIENTS
 Currently, the following are supported:
 .IP \(bu 3
-Brave
+Brave (stable, beta, dev and nightly)
 .IP \(bu 3
 Chromium (stable,beta,dev)
 .IP \(bu 3
diff -Naur a/README.md b/README.md
--- a/README.md	2020-05-16 12:14:59.033740280 -0400
+++ b/README.md	2020-05-16 12:21:00.170292346 -0400
@@ -3,7 +3,7 @@
 
 ## Supported Browsers
 * Aurora
-* Brave (https://aur.archlinux.org/packages/brave-bin)
+* Brave (stable, beta, dev and nightly) (https://aur.archlinux.org/packages/brave-bin)
 * Chromium (stable, beta, and dev)
 * Conkeror
 * Falkon
