--- libs/Makefile.old	2012-10-02 11:25:41.608862450 +0200
+++ libs/Makefile	2012-10-02 11:26:07.139043080 +0200
@@ -20,7 +20,7 @@
 
 # Modify as required.
 #
-PREFIX = /usr/local
+PREFIX = /usr
 SUFFIX := $(shell uname -m | sed -e 's/^unknown/$//' -e 's/^i.86/$//' -e 's/^x86_64/$/64/')
 LIBDIR = lib$(SUFFIX)
 
@@ -31,7 +31,7 @@
 
 
 CPPFLAGS += -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS
-CXXFLAGS += -O2 -Wall -I. -fPIC
+CXXFLAGS += -Wall -I. -fPIC
 LDLFAGS +=
 LDLIBS +=
 
@@ -45,18 +45,14 @@
 
 
 $(ZITA-ALSA-PCMI_MIN): $(ZITA-ALSA-PCMI_O)
-	g++ -shared $(LDFLAGS) -Wl,-soname,$(ZITA-ALSA-PCMI_MAJ) -o $(ZITA-ALSA-PCMI_MIN) $(ZITA-ALSA-PCMI_O) $(ZITA-ALSA-PCMI_DEP)
+	$(CXX) -shared $(LDFLAGS) -Wl,-soname,$(ZITA-ALSA-PCMI_MAJ) -o $(ZITA-ALSA-PCMI_MIN) $(ZITA-ALSA-PCMI_O) $(ZITA-ALSA-PCMI_DEP)
 
 
 install:	$(ZITA-ALSA-PCMI_MIN)
 	install -Dm 644 $(ZITA-ALSA-PCMI_H)   $(DESTDIR)$(PREFIX)/include/$(ZITA-ALSA-PCMI_H)
 	install -Dm 644 $(ZITA-ALSA-PCMI_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(ZITA-ALSA-PCMI_MIN)
-	ln -sf $(ZITA-ALSA-PCMI_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(ZITA-ALSA-PCMI_SO)
-	ldconfig
-
-uninstall:
-	rm -rf $(DESTDIR)$(PREFIX)/include/zita-alsa-pcmi.h
-	rm -rf $(DESTDIR)$(PREFIX)/$(LIBDIR)/libzita-alsa-pcmi*
+	ln -sf $(ZITA-ALSA-PCMI_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(ZITA-ALSA-PCMI_SO)
+	ln -sf $(ZITA-ALSA-PCMI_MIN) $(DESTDIR)$(PREFIX)/$(LIBDIR)/$(ZITA-ALSA-PCMI_MAJ)
 
 clean:
 	/bin/rm -f *~ *.o *.a *.d *.so.*
--- apps/Makefile.orig	2012-11-01 00:50:06.272961455 +0000
+++ apps/Makefile	2012-11-01 00:53:26.586873134 +0000
@@ -24,25 +24,25 @@
 LIBDIR = lib$(SUFFIX)
 VERSION = "0.1.0"
 LDFLAGS += -L$(PREFIX)/$(LIBDIR)
-CPPFLAGS += -DVERSION=\"$(VERSION)\"
-CXXFLAGS += -O2 -Wall -MMD -MP
+CPPFLAGS += -DVERSION=\"$(VERSION)\" -I../libs
+CXXFLAGS += -Wall -MMD -MP
 
 
 all:	alsa_loopback alsa_delay
 
 
 ALSA_LOOPBACK_O =	alsa_loopback.o pxthread.o
-alsa_loopback:	LDLIBS += -lzita-alsa-pcmi -lasound -lpthread -lrt
+alsa_loopback:	LDLIBS += ../libs/libzita-alsa-pcmi.so.0.2.0 -lasound -lpthread -lrt
 alsa_loopback:	$(ALSA_LOOPBACK_O)
-	g++ $(LDFLAGS) -o $@ $(ALSA_LOOPBACK_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(ALSA_LOOPBACK_O) $(LDLIBS)
 $(ALSA_LOOPBACK_O):
 -include $(_ALSA_LOOPBACK_O:%.o=%.d)
 
 
 ALSA_DELAY_O =	alsa_delay.o mtdm.o pxthread.o
-alsa_delay:	LDLIBS += -lzita-alsa-pcmi -lasound -lpthread -lrt
+alsa_delay:	LDLIBS += ../libs/libzita-alsa-pcmi.so.0.2.0 -lasound -lpthread -lrt
 alsa_delay:	$(ALSA_DELAY_O)
-	g++ $(LDFLAGS) -o $@ $(ALSA_DELAY_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(ALSA_DELAY_O) $(LDLIBS)
 $(ALSA_DELAY_O):
 -include $(ALSA_DELAY_O:%.o=%.d)
 
