--- a/haguichi-1.3.1/cmake/FindVala.cmake	2015-12-30 15:41:48.189227095 -0300
+++ b/haguichi-1.3.0/cmake/FindVala.cmake	2015-12-30 15:41:25.776954585 -0300
@@ -45,7 +45,7 @@
 
 
 # Search for the valac executable in the usual system paths.
-find_program(VALA_EXECUTABLE NAMES valac)
+find_program(VALA_EXECUTABLE NAMES valac valac-0.26 valac-0.28 valac-0.30)
 
 if (VALA_EXECUTABLE STREQUAL "VALA_EXECUTABLE-NOTFOUND")
 
