--- a/flowblade-trunk/flowblade	2016-09-21 22:45:02.443196996 -0300
+++ b/flowblade-trunk/flowblade	2016-09-21 22:46:04.333198137 -0300
@@ -35,9 +35,7 @@
 # Update sys.path to include modules
 if launch_dir == "/usr/bin":
     print "Running from installation..."
-    modules_path = "/usr/share/flowblade/Flowblade"
-    if not os.path.isdir(modules_path):
-        modules_path = "/usr/share/pyshared/Flowblade"
+    modules_path = "/usr/share/flowblade"
     print "modules path:", modules_path
 else:
     print "Running from filesystem..."
