From 43080e62a15541c913ddeeb06e085796998ec94a Mon Sep 17 00:00:00 2001
From: devtrium <devtrium325@gmail.com>
Date: Sun, 9 Jul 2017 14:41:31 +0800
Subject: [PATCH] add snazzy color scheme

---
 src/tilda.ui |  3 +++
 src/wizard.c | 32 +++++++++++++++++++++++++++++++-
 2 files changed, 34 insertions(+), 1 deletion(-)

diff --git a/src/tilda.ui b/src/tilda.ui
index 4c4aa6b..656396f 100644
--- a/src/tilda.ui
+++ b/src/tilda.ui
@@ -214,6 +214,9 @@
       <row>
         <col id="0" translatable="yes">Solarized Dark</col>
       </row>
+      <row>
+        <col id="0" translatable="yes">Snazzy</col>
+      </row>
     </data>
   </object>
   <object class="GtkListStore" id="model6">
diff --git a/src/wizard.c b/src/wizard.c
index 4e43875..580fe41 100644
--- a/src/wizard.c
+++ b/src/wizard.c
@@ -189,6 +189,26 @@ terminal_palette_solarizedD[TERMINAL_PALETTE_SIZE] = {
 	{ RGB(0xfdfd, 0xf6f6, 0xe3e3) }
 };
 
+const GdkRGBA
+terminal_palette_snazzy[TERMINAL_PALETTE_SIZE] = {
+	{ RGB(0x2828, 0x2a2a, 0x3636) },
+	{ RGB(0xffff, 0x5c5c, 0x5757) },
+	{ RGB(0x5a5a, 0xf7f7, 0x8e8e) },
+	{ RGB(0xf3f3, 0xf9f9, 0x9d9d) },
+	{ RGB(0x5757, 0xc7c7, 0xffff) },
+	{ RGB(0xffff, 0x6a6a, 0xc1c1) },
+	{ RGB(0x9a9a, 0xeded, 0xfefe) },
+	{ RGB(0xf1f1, 0xf1f1, 0xf0f0) },
+	{ RGB(0x6868, 0x6868, 0x6868) },
+	{ RGB(0xffff, 0x5c5c, 0x5757) },
+	{ RGB(0x5a5a, 0xf7f7, 0x8e8e) },
+	{ RGB(0xf3f3, 0xf9f9, 0x9d9d) },
+	{ RGB(0x5757, 0xc7c7, 0xffff) },
+	{ RGB(0xffff, 0x6a6a, 0xc1c1) },
+	{ RGB(0x9a9a, 0xeded, 0xfefe) },
+	{ RGB(0xf1f1, 0xf1f1, 0xf0f0) }
+};
+
 typedef struct _TerminalPaletteScheme
 {
   const char *name;
@@ -203,7 +223,8 @@ static TerminalPaletteScheme palette_schemes[] = {
     { N_("Rxvt"), terminal_palette_rxvt },
     { N_("Zenburn"), terminal_palette_zenburn },
     { N_("Solarized Light"), terminal_palette_solarizedL },
-    { N_("Solarized Dark"), terminal_palette_solarizedD }
+    { N_("Solarized Dark"), terminal_palette_solarizedD },
+    { N_("Snazzy"), terminal_palette_snazzy }
 };
 
 /* For use in get_display_dimension() */
@@ -1472,6 +1493,15 @@ static void combo_colorschemes_changed_cb (GtkWidget *w, tilda_window *tw)
 			gdk_back.green = 0.17;
 			gdk_back.blue = 0.21;
 			break;
+		/* Snazzy */
+		case 7:
+			gdk_text.red = 0.94;
+			gdk_text.green = 0.94;
+			gdk_text.blue = 0.92;
+			gdk_back.red = 0.16;
+			gdk_back.green = 0.16;
+			gdk_back.blue = 0.21;
+			break;
 	    /* Custom */
         default:
             nochange = TRUE;
