--- app/Makefile.am.org	2015-03-30 17:29:46.025117693 +0000
+++ app/Makefile.am	2015-03-30 17:29:56.905118080 +0000
@@ -29,16 +29,6 @@
 	@echo "Installing the scripter..."
 	@ln -s gbs$(GAMBAS_VERSION).gambas $(DESTDIR)$(bindir)/gbs$(GAMBAS_VERSION) || true
 	@ln -s gbs$(GAMBAS_VERSION).gambas $(DESTDIR)$(bindir)/gbw$(GAMBAS_VERSION) || true
-	@if test x"$(XDG_UTILS)" != x; then \
-		$(INSTALL) -d $(DESTDIR)$(gbdatadir)/icons; \
-		cp -f $(srcdir)/mime/application-x-gambas*.png $(DESTDIR)$(gbdatadir)/icons; \
-		echo "Registering Gambas script mimetype"; \
-		xdg-icon-resource install --context mimetypes --size 64 $(DESTDIR)$(gbdatadir)/icons/application-x-gambasscript.png application-x-gambasscript; \
-		xdg-mime install $(srcdir)/mime/application-x-gambasscript.xml; \
-		echo "Registering Gambas server page mimetype"; \
-		xdg-icon-resource install --context mimetypes --size 64 $(DESTDIR)$(gbdatadir)/icons/application-x-gambasserverpage.png application-x-gambasserverpage; \
-		xdg-mime install $(srcdir)/mime/application-x-gambasserverpage.xml; \
-	fi
 
 	@echo "Installing the Gambas appdata file"
 	@$(INSTALL) -d $(DESTDIR)$(datarootdir)/appdata
