diff -Naur a/app/Makefile.am b/app/Makefile.am
--- a/app/Makefile.am	2015-10-04 11:04:11.000000000 -0300
+++ b/app/Makefile.am	2015-11-04 04:31:59.390153512 -0300
@@ -29,16 +29,8 @@
 	@echo "Installing the scripter..."
 	@ln -s gbs$(GAMBAS_VERSION).gambas $(DESTDIR)$(bindir)/gbs$(GAMBAS_VERSION) || true
 	@ln -s gbs$(GAMBAS_VERSION).gambas $(DESTDIR)$(bindir)/gbw$(GAMBAS_VERSION) || true
-	@if test x"$(XDG_UTILS)" != x; then \
-		$(INSTALL) -d $(DESTDIR)$(gbdatadir)/icons; \
-		cp -f $(srcdir)/mime/application-x-gambas*.png $(DESTDIR)$(gbdatadir)/icons; \
-		echo "Registering Gambas script mimetype"; \
-		xdg-icon-resource install --context mimetypes --size 256 $(DESTDIR)$(gbdatadir)/icons/application-x-gambasscript.png application-x-gambasscript; \
-		xdg-mime install $(srcdir)/mime/application-x-gambasscript.xml; \
-		echo "Registering Gambas server page mimetype"; \
-		xdg-icon-resource install --context mimetypes --size 256 $(DESTDIR)$(gbdatadir)/icons/application-x-gambasserverpage.png application-x-gambasserverpage; \
-		xdg-mime install $(srcdir)/mime/application-x-gambasserverpage.xml; \
-	fi
+	$(INSTALL) -d $(DESTDIR)$(gbdatadir)/icons;
+	cp -f $(srcdir)/mime/application-x-gambas*.png $(DESTDIR)$(gbdatadir)/icons;
 
 	@echo "Installing the Gambas appdata file"
 	@$(INSTALL) -d $(DESTDIR)$(datarootdir)/appdata
diff -Naur a/main/Makefile.am b/main/Makefile.am
--- a/main/Makefile.am	2015-10-04 11:04:28.000000000 -0300
+++ b/main/Makefile.am	2015-11-04 04:34:34.560674138 -0300
@@ -28,13 +28,9 @@
 	@echo "Making runtime symbolic link"
 	@$(LN_S) -f gbx$(GAMBAS_VERSION) $(DESTDIR)$(bindir)/gbr$(GAMBAS_VERSION) || true
 	
-	@if test x"$(XDG_UTILS)" != x; then \
-		echo "Registering Gambas executable mimetype"; \
-		$(INSTALL) -d $(DESTDIR)$(gbdatadir)/icons; \
-		cp -f $(srcdir)/mime/application-x-gambas3.png $(DESTDIR)$(gbdatadir)/icons; \
-		xdg-icon-resource install --context mimetypes --size 256 $(DESTDIR)$(gbdatadir)/icons/application-x-gambas3.png application-x-gambas3; \
-		xdg-mime install $(srcdir)/mime/application-x-gambas3.xml; \
-	fi
+	$(INSTALL) -d $(DESTDIR)$(gbdatadir)/icons;
+	cp -f $(srcdir)/mime/application-x-gambas3.png $(DESTDIR)$(gbdatadir)/icons;
+
 
 	@echo "Creating the information files for gb component..."
 	@$(INSTALL) -d $(DESTDIR)$(gbdatadir)/info
