From e0923a04940e04ea7af5985f08a0375e80ad5b85 Mon Sep 17 00:00:00 2001
From: Jonas Kvinge <jonas@jkvinge.net>
Date: Tue, 23 Apr 2019 23:03:59 +0200
Subject: [PATCH] Fix small sidebar mode width

---
 src/widgets/fancytabwidget.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/widgets/fancytabwidget.cpp b/src/widgets/fancytabwidget.cpp
index a593222..e5741be 100644
--- a/src/widgets/fancytabwidget.cpp
+++ b/src/widgets/fancytabwidget.cpp
@@ -90,9 +90,10 @@ class FancyTabBar: public QTabBar {
     FancyTabWidget *tabWidget = (FancyTabWidget*) parentWidget();
     QSize size = FancyTabWidget::TabSize_LargeSidebar;
 
-    if(tabWidget->mode() != FancyTabWidget::Mode_LargeSidebar) {
+    if (tabWidget->mode() != FancyTabWidget::Mode_LargeSidebar) {
       size = QTabBar::tabSizeHint(index);
     }
+    size.setWidth(std::max(size.width(), 37));
 
     return size;
 
