--- a/configure.ac	2013-04-02 10:19:10.000000000 -0300
+++ b/configure.ac	2017-02-25 22:03:33.920744317 -0300
@@ -117,9 +117,9 @@
   AC_CHECK_LIB(c_r,pthread_create,LIBPTHREAD="-lc_r")
 fi
 AC_CHECK_LIB(ossaudio,main,LIBOSS="-lossaudio")
-AC_CHECK_LIB(ncursesw,initscr,LIBCURSES="-lncursesw")
+AC_CHECK_LIB(ncursesw,initscr,LIBCURSES="-lncursesw -ltinfow")
 if test "$LIBCURSES" = ""; then
-  AC_CHECK_LIB(ncurses,initscr,LIBCURSES="-lncurses")
+  AC_CHECK_LIB(ncurses,initscr,LIBCURSES="-lncurses -ltinfo")
 fi
 if test "$LIBCURSES" = ""; then
   AC_CHECK_LIB(curses,initscr,LIBCURSES="-lcurses")
@@ -131,7 +131,7 @@
   exit 1
 fi
 if test "$LIBCURSES" = "-lncursesw"; then
-  CFLAGS="$CFLAGS -I/usr/include/ncursesw"
+  CFLAGS="$CFLAGS -D_GNU_SOURCE -I/usr/include/ncursesw"
 fi
 AC_SUBST(LIBPTHREAD)
 AC_SUBST(LIBOSS)
