--- source/Makefile.orig	2012-11-01 01:30:56.224700444 +0000
+++ source/Makefile	2012-11-01 01:37:42.013382428 +0000
@@ -25,7 +25,7 @@
 LIBDIR = lib$(SUFFIX)
 LDFLAGS += -L$(PREFIX)/$(LIBDIR)
 CPPFLAGS += -DVERSION=\"$(VERSION)\"
-CXXFLAGS += -O2 -Wall -MMD -MP
+CXXFLAGS += -Wall -MMD -MP
 
 
 all:	zita-a2j zita-j2a
@@ -37,7 +37,7 @@
 zita-a2j:	CPPFLAGS += -DAPPNAME=\"zita-a2j\"
 zita-a2j:	LDLIBS += -lzita-resampler -lzita-alsa-pcmi -ljack -lasound -lpthread -lm -lrt
 zita-a2j:	$(ZITA-A2J_O)
-	g++ $(LDFLAGS) -o $@ $(ZITA-A2J_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(ZITA-A2J_O) $(LDLIBS)
 
 
 ZITA-J2A_O = zita-j2a.o alsathread.o jackclient.o pxthread.o lfqueue.o
@@ -46,7 +46,7 @@
 zita-j2a:	CPPFLAGS += -DAPPNAME=\"zita-j2a\"
 zita-j2a:	LDLIBS += -lzita-resampler -lzita-alsa-pcmi -ljack -lasound -lpthread -lm -lrt
 zita-j2a:	$(ZITA-J2A_O)
-	g++ $(LDFLAGS) -o $@ $(ZITA-J2A_O) $(LDLIBS)
+	$(CXX) $(LDFLAGS) -o $@ $(ZITA-J2A_O) $(LDLIBS)
 
 
 install:	all
