diff -Naur a/lib/CMakeLists.txt b/lib/CMakeLists.txt
--- a/lib/CMakeLists.txt	2019-04-14 02:16:04.000162501 -0400
+++ b/lib/CMakeLists.txt	2019-04-14 02:17:52.181809018 -0400
@@ -3,6 +3,9 @@
 #####
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 add_subdirectory(dbusmenu-importer)
+if(${VALA_VERSION} VERSION_LESS "0.40.4")
+    set(ADDONS_PACKAGE gio-addons-2.0)
+endif()
 set(CORE_VALA_FILES
     appmenu-abstractions.vala
     appmenu-bamf.vala
@@ -19,7 +22,7 @@
 PACKAGES
     ${CORE_PACKAGES}
     dbus-menu-importer
-    gio-addons-2.0
+    ${ADDONS_PACKAGE}
     launcher
     ${WNCK_PACKAGES}
 OPTIONS
