diff -Naur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2017-09-09 00:18:21.384826286 -0300
+++ b/src/CMakeLists.txt	2017-09-09 00:21:46.314826345 -0300
@@ -238,7 +238,7 @@
     )
 
     # Configure and install Python extensions for Nautilus-derived file managers
-    foreach(_file_manager Nautilus Nemo Caja)
+    foreach(_file_manager Nautilus Nemo)
         string(TOLOWER ${_file_manager} _file_manager_lower)
         set(_py_filename "${CMAKE_CURRENT_BINARY_DIR}/dist/nitroshare_${_file_manager_lower}.py")
         configure_file("${CMAKE_CURRENT_SOURCE_DIR}/dist/nitroshare.py.in" "${_py_filename}")
@@ -247,4 +247,14 @@
             RENAME nitroshare.py
         )
     endforeach()
+
+    #Install in Caja
+    string(TOLOWER Caja _caja_lower)
+    set(_py_filename "${CMAKE_CURRENT_BINARY_DIR}/dist/nitroshare_${_caja_lower}.py")
+    configure_file("${CMAKE_CURRENT_SOURCE_DIR}/dist/nitroshare.py.in" "${_py_filename}")
+    install(FILES "${_py_filename}"
+        DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/python-${_caja_lower}/extensions"
+         RENAME nitroshare.py
+    )
+
 endif()
