--- a/caffeine/caffeine-screensaver	2015-01-16 14:45:37.000000000 -0300
+++ b/caffeine/caffeine-screensaver 2015-06-04 16:13:34.024774925 -0300
@@ -336,6 +336,7 @@
     elif `dbus-send --print-reply --dest=org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus.GetNameOwner string:org.gnome.SessionManager > /dev/null 2>&1` ; then DE=gnome;
     elif xprop -root _DT_SAVE_MODE 2> /dev/null | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
     elif [ x"$DESKTOP_SESSION" = x"LXDE" ]; then DE=lxde;
+	elif [ -n "$MATE_DESKTOP_SESSION_ID" ]; then DE=mate;
     else DE=""
     fi
 }
@@ -439,7 +440,9 @@
     xfce)
       screensaver_xserver "$1"
       ;;
-
+	mate)
+	  screensaver_mate "$1"
+	 ;;
     '')
       screensaver_xserver "$1"
       ;;
@@ -641,6 +644,49 @@
     esac
 }
 
+screensaver_mate(){
+	case "$1" in
+		suspend)
+		mate-screensaver-command --inhibit > /dev/null 2>/dev/null &
+		result=$?
+		;;
+		resume)
+		PID=`pidof mate-screensaver-command`
+		if [ -n "$PID" ];then
+			kill -9 $PID
+		fi
+		result=$?
+		;;
+		activate)
+		mate-screensaver-command --activate > /dev/null 2>/dev/null
+		result=$?
+		;;
+		lock)
+		mate-screensaver-command --lock > /dev/null 2>/dev/null
+		result=$?
+		;;
+		reset)
+		mate-screensaver-command --deactivate > /dev/null 2>/dev/null
+		result=$?
+		;;
+		status)
+		mate-screensaver-command --query | grep inactiv > /dev/null 2>/dev/null
+		result=$?
+		if [ "$result" -eq 1 ];then
+			echo "enabled"
+		elif [ "$result" -eq 0 ];then
+			echo "disabled"
+		else
+			echo "ERROR: could not be queried"
+		fi
+		;;
+		*)
+		echo "ERROR: Unknown command '$1'" >&2
+        return 1
+		;;
+	esac
+}
+
 screensaver_xserver()
 {
     case "$1" in